In [1]:
import math

print(math.pi)

3.141592653589793


In [2]:
from modules.data_preprocessing import clean_data

clean_data()

Cleaned data.


### Module Exercise

In [4]:
import json

with open('./modules/data.json','r') as file:
    data = json.load(file)

print(data)


{'name': 'Jared', 'age': 23}


In [1]:
import numpy as np
import pandas as pd

array = np.array([1, 2, 3, 4, 5])

array

array([1, 2, 3, 4, 5])

In [2]:
data = {'ID': [1, 2, 3, 4],
        'Address': ['a', 'b', 'c', 'd'],
        'Sales': [123, 456, 789, 100]}

df = pd.DataFrame(data)
print(df)

   ID Address  Sales
0   1       a    123
1   2       b    456
2   3       c    789
3   4       d    100


In [8]:
average = df['Sales'].mean()

print(average)

367.0


In [9]:
def add(a,b):
    """
    Add two numbers together.

    Parameters:
    a (int): First Number
    b (int): Second Number

    Returns:
    int: Sum of the two numbers
    """
    return a + b

help(add)

Help on function add in module __main__:

add(a, b)
    Add two numbers together.

    Parameters:
    a (int): First Number
    b (int): Second Number

    Returns:
    int: Sum of the two numbers



In [13]:
def multiply(a, b):
    """
    Multiply two numbers together.

    Parameters:
    a (int): First Number
    b (int): Second Number

    Returns:
    int: Product of the two numbers
    """
    return a * b


print(multiply(3, 7))
help(multiply)

21
Help on function multiply in module __main__:

multiply(a, b)
    Multiply two numbers together.

    Parameters:
    a (int): First Number
    b (int): Second Number

    Returns:
    int: Product of the two numbers



In [21]:
import numpy as np

# 1D ndarray
array1 = np.array([1, 2, 3])
print(array1.shape)

# 2D ndarray
array2 = np.array([[1, 2, 3], [1, 2, 3]])
print(array2.shape)

# Type coercion
array3 = np.array([3.14, 2, 3])
print(array3) # converts everything to float64
print(type(array3[1]))

# Explicit Type Coercion
array4 = np.array([3.14, 2, 3], dtype=np.int64) # dtype = data type for array
print(array4)

(3,)
(2, 3)
[3.14 2.   3.  ]
<class 'numpy.float64'>
[3 2 3]


In [1]:
import numpy as np

a = np.array([10,20,30,40])
b = np.arange(4) # create array from 0 to 3

d = a + b
print(d)
d = a - b
print(d)
d = a * b
print(d)
d = np.array(b * a / a, dtype=np.int64)
print(d)

[10 21 32 43]
[10 19 28 37]
[  0  20  60 120]
[0 1 2 3]


In [2]:
# Basic Array Operations
A = np.array([[1,1], [0,1]])
B = np.array([[2,0], [3,4]])

print(A*B)

[[2 0]
 [0 4]]


In [11]:
arr = np.arange(10)**2
print(arr)
print(np.array([i ** 2 for i in range(10)]))
arr[0:5]

[ 0  1  4  9 16 25 36 49 64 81]
[ 0  1  4  9 16 25 36 49 64 81]


array([ 0,  1,  4,  9, 16])

In [13]:
A = np.array([[[1, 1, 3], [9, 8, 7], [5, 1, 3]], [[1, 1, 3], [9, 8, 7], [
             50, 1, 3]], [[10, 10, 30], [90, 80, 70], [500, 10, 300]]])
print(A.shape)
print(A)

(3, 3, 3)
[[[  1   1   3]
  [  9   8   7]
  [  5   1   3]]

 [[  1   1   3]
  [  9   8   7]
  [ 50   1   3]]

 [[ 10  10  30]
  [ 90  80  70]
  [500  10 300]]]


In [19]:
# The first index inside the third array of the second array
print(A[1,2,0])
print(A[1][2][0])
A[0:2]

50
50


array([[[ 1,  1,  3],
        [ 9,  8,  7],
        [ 5,  1,  3]],

       [[ 1,  1,  3],
        [ 9,  8,  7],
        [50,  1,  3]]])

In [21]:
sequence = np.arange(0,30,2) # start, stop, steps
sequence

array([ 0,  2,  4,  6,  8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28])

In [24]:
np.linspace(0,30,5) # start, stop, no. of points (divides range into no. of points)

array([ 0. ,  7.5, 15. , 22.5, 30. ])

In [30]:
x = np.random.random((2,3)) # takes in a shape, and generates a random float between 0 and 1
x

array([[0.54379504, 0.99029757, 0.46085687],
       [0.18133611, 0.41406711, 0.47812239]])

In [36]:
x = np.random.randint(0,100,(3,3)) # min, max, shape
x

array([[16, 20, 42],
       [ 6, 23, 39],
       [20, 50, 71]])

### Basic Statistics

In [47]:
data = np.random.randint(75, 100, (3,3))
data

array([[79, 79, 98],
       [84, 75, 94],
       [80, 85, 94]])

In [48]:
np.mean(data)

85.33333333333333

In [49]:
np.median(data)

84.0

In [50]:
np.std(data)

7.659416862050705

In [52]:
import math
math.sqrt(78.56)

8.863407922464136

In [54]:
data = np.random.randint(75, 100, (5))
data

array([92, 97, 99, 91, 95])

In [59]:
print(np.max(data))
print(np.min(data))
print(np.sum(data))
print(np.mean(data))
print(np.std(data))
print(np.var(data))

99
91
474
94.8
2.993325909419153
8.959999999999999


In [71]:
import numpy as np

np.random.seed(42) # pick an id for a random set of numbers

survey_data = np.random.randint(1,6,size= (10,4))
print(survey_data)

categories = ["Work Environment", "Salary Satisfaction", "Work-life Balance", "Career Development Opportunities"]
categories

average = np.mean(survey_data, axis=0)
overall_average = np.mean(survey_data)
categories[np.argmax(average)] # get category with highest
categories[np.argmin(average)] # with lowest
print(np.max(average))
print(np.min(average))
print(overall_average)
np.std(survey_data, axis=0) # How varied the data is


[[4 5 3 5]
 [5 2 3 3]
 [3 5 4 3]
 [5 2 4 2]
 [4 5 1 4]
 [2 5 4 1]
 [1 3 3 2]
 [4 4 3 4]
 [4 1 3 5]
 [3 5 1 2]]
3.7
2.9
3.3


array([1.20415946, 1.48660687, 1.04403065, 1.3       ])

### Interpretation of the Result

1. Salary satisfaction has the highest average score of 3.7, indicating that, on average, employees feel relatively more satisfied with their salaries than with other aspects covered in the survey.

2. Work-life balance has the lowest average score of 2.9, indicating that, on average, employees are having a harder time in balancing professional and personal lives.

3. The overall Average Satisfaction Score across all categories is 2.96, which is satisfactory with room for improvements for the company.